polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -1107,12 +1210,13 @@ void LoopInternalization::transform(T loop,
       offset = ValueOrUnsigned::add(offset, reqdLocalMemory, builder);
     }
   }
-  LLVM_DEBUG(llvm::dbgs() << "Promoted loop: " << loop << "\n");
 
   builder.setInsertionPoint(loop);
   createLocalBarrier(builder);
   builder.setInsertionPointAfter(loop);
   createLocalBarrier(builder);
+
+  LLVM_DEBUG(llvm::dbgs() << "Promoted loop: " << loop << "\n");
```

Comment:
I considered adding barrier not part of promotion, and it doesn't change the loop, why moving the debug line down here?

Reply:
I moved it because I wanted to see the loop after the transformation is complete but I can move it back if you feel it should go in the original place.